install_with_binary.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Function to display step messages
+display_step() {
+    echo "Step $1: $2"
+}
+
+# Step 1: Detect the current OS and Architecture type
+display_step 1 "Detecting current OS and Arch "
+os=""
+case "$(uname -sr)" in
+   Darwin*)
+     os="darwin" ;;
+   Ubuntu*|Pop!_OS)
+     os="linux" ;;
+   Linux*amzn2*)
+     os="amazon linux" ;;
+   Linux*)
+     os="linux" ;;
+   Debian*)
+     os="linux" ;;
+   Linux\ Mint*)
+     os="linux" ;;
+   Red\ Hat*)
+     os="linux" ;;
+   CentOS*)
+     os="linux" ;;
+   *)
+     os="Not Found: $os_name"
```

Comment:
I don't see `os_name` defined anywhere. Is it set somehow by this switch statement?